streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorNode.java

```diff
@@ -66,10 +67,11 @@ public void run() {
     };
 
     private long timestamp;
+    private Punctuator punctuator;
```

Comment:
I assumed it to be an optimisation - punctuateDelegate is constructed once (like the other delegates) and timestamp and punctuator fields are updated in place before each time the delegate uses them from its run() method